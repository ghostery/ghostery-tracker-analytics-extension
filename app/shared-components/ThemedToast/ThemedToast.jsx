import React from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';

import './ThemedToast.scss';

const ThemedToast = ({
  show, initialLoad, toastText, actions, altStyling, errorStyling, panel,
}) => {
  const toastClasses = ClassNames(
    'ThemedToast',
    'd-flex',
    'justify-content-between',
    'align-items-center',
    {
      show,
      initialLoad,
      'ThemedToast--small': altStyling,
      changePosition: altStyling && panel,
      error: errorStyling,
    },
  );

  const closeButtonClasses = ClassNames(
    'ThemedToast__closeButton',
    { error: altStyling },
  );

  return (
    <div className={toastClasses}>
      <div className="d-flex">
        {toastText === 'You are now in Freeze mode. Data is not populating in real time.'
          && (<p className="ThemedToast--boldText">Heads up!</p>)}
        <p>{toastText}</p>
      </div>
      <img
        alt="Close Toast"
        src={[chrome.extension.getURL('dist/images/shared/x-icon-semi-transparent.svg')]}
        className={closeButtonClasses}
        onClick={actions.closeToast}
      />
    </div>
  );
};

ThemedToast.propTypes = {
  show: PropTypes.bool.isRequired,
  toastText: PropTypes.string.isRequired,
  initialLoad: PropTypes.bool.isRequired,
  actions: PropTypes.shape({
    closeToast: PropTypes.func.isRequired,
  }).isRequired,
  altStyling: PropTypes.bool,
  errorStyling: PropTypes.bool,
  panel: PropTypes.bool,
};

ThemedToast.defaultProps = {
  altStyling: false,
  errorStyling: false,
  panel: false,
};

export default ThemedToast;
