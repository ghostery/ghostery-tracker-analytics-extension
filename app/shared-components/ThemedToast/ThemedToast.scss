@import '../../utils/scss/color-variables.scss';
@import '../../utils/scss/z-index-variables.scss';

.ThemedToast {
  visibility: visible;
  position: absolute;
  z-index: $z-index-toast;
  top: 0;
  left: 50%;
  margin-left: -300px;
  opacity: 0;
  box-shadow: inset 1px 0 6px $opaque-background-medium;
  background-color: $light-blue;
  border-radius: 5px;
  padding: 20px;
  min-width: 600px;
  font-weight: 600;
  color: $relaxed-blue-darker;
  transition: top .5s, opacity .5s;

  &.initialLoad {
    visibility: hidden;
  }

  p {
    margin: 2px;
    font-size: 14px;
  }

  &.show {
    top: 30px;
    opacity: 1;
    transition: top .5s, opacity .5s;
  }

  &:not(.show) {
    pointer-events: none;
  }
}

.ThemedToast--boldText {
  font-weight: 700;
}

.ThemedToast__closeButton {
  width: 9px;
  height: 10px;
  cursor: pointer;

  &.error {
    position: relative;
    left: 4px;
  }
}

.ThemedToast--small {
  position: absolute;
  left: unset;
  right: -5px;
  box-shadow: unset;
  padding: 10px 20px;
  width: 235px;
  min-width: unset;
  font-weight: 700;

  p {
    font-size: 11px;
  }

  &.show {
    top: 20px;
  }

  &.changePosition {
    position: fixed;
    left: unset;
    right: 12px;
    text-align: center;
  }

  &.error {
    background-color: $light-red;
    color: $info-red;
  }
}
