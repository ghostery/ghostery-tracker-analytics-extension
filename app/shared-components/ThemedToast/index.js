import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import ThemedToast from './ThemedToast.jsx';
import { closeToast } from '../../store/actions/UserInterfaceActions';

const mapStateToProps = state => ({
  toastText: state.UserInterfaceReducer.toast.toastText,
  show: state.UserInterfaceReducer.toast.show,
  initialLoad: state.UserInterfaceReducer.toast.initialLoad,
  altStyling: state.UserInterfaceReducer.toast.altStyling,
  errorStyling: state.UserInterfaceReducer.toast.errorStyling,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ closeToast }, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(ThemedToast);
