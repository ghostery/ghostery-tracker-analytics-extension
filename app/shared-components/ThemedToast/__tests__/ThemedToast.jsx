import React from 'react';
import { shallow } from 'enzyme';
import chrome from 'sinon-chrome';
import path from 'path';
import fakeProps from 'react-fake-props';

import ThemedToast from '../ThemedToast';

jest.mock('../../../../src/classes/Globals', () => ({
  BROWSER_INFO: { displayName: 'Chrome' },
}));

describe('app/shared-components/ThemedToast/', () => {
  describe('Snapshot test with shallow', () => {
    test('ThemedToast should render', () => {
      global.chrome = chrome;
      const componentPath = path.join(__dirname, '../ThemedToast.jsx');
      const props = fakeProps(componentPath);

      const component = shallow(<ThemedToast {...props} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('Functionality testing', () => {
    test('onClick should be called', () => {
      global.chrome = chrome;
      const mockCallBack = jest.fn();
      const props = {
        show: true,
        toastText: 'testing',
        initialLoad: true,
        actions: {
          closeToast: mockCallBack,
        },
        altStyling: true,
        errorStyling: true,
        panel: true,
      };

      const component = shallow(<ThemedToast {...props} onClick={mockCallBack} />);

      component.find('img.ThemedToast__closeButton').simulate('click');
      expect(mockCallBack.mock.calls.length).toEqual(1);
    });
  });
});
