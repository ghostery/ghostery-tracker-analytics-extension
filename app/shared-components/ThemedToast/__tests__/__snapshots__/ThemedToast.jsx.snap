// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/shared-components/ThemedToast/ Snapshot test with shallow ThemedToast should render 1`] = `
<div
  className="ThemedToast d-flex justify-content-between align-items-center show initialLoad"
>
  <div
    className="d-flex"
  >
    <p>
      toastText
    </p>
  </div>
  <img
    alt="Close Toast"
    className="ThemedToast__closeButton"
    onClick={[Function]}
    src={
      Array [
        undefined,
      ]
    }
  />
</div>
`;
